tests/unit/src/main/scala/tests/CodeLensesTextEdits.scala

```diff
@@ -3,18 +3,21 @@ package tests
 import org.eclipse.{lsp4j => l}
 
 object CodeLensesTextEdits {
-  def apply(ranges: Seq[l.CodeLens]): List[l.TextEdit] = {
-    ranges.map(convert).toList
+  def apply(lenses: Seq[l.CodeLens]): Iterable[l.TextEdit] = {
+    for {
+      (line, lenses) <- lenses.groupBy(_.getRange.getStart.getLine)
+      textEdit = convert(line, lenses)
+    } yield textEdit
   }
 
-  private def convert(lens: l.CodeLens): l.TextEdit = {
-    val range = lens.getRange
-    val pos = new l.Position(range.getStart.getLine, 0)
-    new l.TextEdit(new l.Range(pos, pos), print(lens))
+  private def convert(line: Int, lenses: Seq[l.CodeLens]): l.TextEdit = {
+    val pos = new l.Position(line, 0)
+    val edit = lenses.map(print).mkString
+    new l.TextEdit(new l.Range(pos, pos), edit + "\n")
```

Comment:
Will it not be a problem on windows by any chance?

Reply:
Changed to `System.lineSeparator`